spark/src/main/scala/geotrellis/spark/io/accumulo/AccumuloWriteStrategy.scala

```diff
@@ -0,0 +1,121 @@
+ *
+ * Note: Giving relative URLs will cause HDFS to use the `fs.defaultFS` property in `core-site.xml`. 
+ * If not specified this will default to local ('file:/') system, this is undesriable.
+ *
+ * @param ingestPath Path where spark will write RDD records for ingest
+ */
+case class HdfsWriteStrategy(ingestPath: Path) extends AccumuloWriteStrategy {
+  def write(kvPairs: RDD[(Key, Value)], instance: AccumuloInstance, table: String): Unit = {
+    val sc = kvPairs.sparkContext
+    val job = Job.getInstance(sc.hadoopConfiguration)
+    instance.setAccumuloConfig(job)
+    val conf = job.getConfiguration
+    val outPath = HdfsUtils.tmpPath(ingestPath, UUID.randomUUID.toString, conf)
+    val failuresPath = outPath.suffix("-failures")
+
+    HdfsUtils.ensurePathExists(failuresPath, conf)
+    kvPairs
+      .sortBy{ case (key, _) => key }
+      .saveAsNewAPIHadoopFile(
+        outPath.toString,
+        classOf[Key],
+        classOf[Value],
+        classOf[AccumuloFileOutputFormat],
+        conf)
+
+    val ops = instance.connector.tableOperations()
+    ops.importDirectory(table, outPath.toString, failuresPath.toString, true)
+
+    // cleanup ingest directories on success
+    val fs = ingestPath.getFileSystem(conf)
+    if( fs.exists(new Path(outPath, "_SUCCESS")) ) {
+      fs.delete(outPath, true)
+      fs.delete(failuresPath, true)
+    } else {
+      throw new java.io.IOException(s"Accumulo bulk ingest failed at $ingestPath")
+    }
+  }
+}
+object HdfsWriteStrategy {
+  def apply(ingestPath: String): HdfsWriteStrategy = HdfsWriteStrategy(new Path(ingestPath))
+}
+
+/**
+ * This strategy will create one BatchWriter per partition and attempt to stream the records to the target tablets.
+ * In order to gain some parallism this strategy will create a number of splits in the target table equal to the number
+ * of tservers in the cluster. This is suitable for smaller ingests, or where HdfsWriteStrategy is otherwise not possible.
+ * 
+ * There is a problem in Accumulo 1.6 (fixed in 1.7) where the split creation does not wait for the resulting
+ * empty tablets to distribute through the cluster before returning. This will create a warm-up period where the
+ * pressure the ingest writers on that node will delay tablet re-balancing.
+ *
+ * The speed of the ingest can be improved by setting `tserver.wal.sync.method=hflush` in accumulo shell.
+ * Note: this introduces higher chance of data loss due to sudden node failure.
+ *
+ * BatchWriter is notified of the tablet migrations and will follow them around the cluster.
+ *
+ * @param config Configuration for the BatchWriters
+ */
+case class SocketWriteStrategy(
+  config: BatchWriterConfig = new BatchWriterConfig().setMaxMemory(128*1024*1024).setMaxWriteThreads(32) 
+) extends AccumuloWriteStrategy {
+  def write(kvPairs: RDD[(Key, Value)], instance: AccumuloInstance, table: String): Unit = {
+    val BC = KryoWrapper((instance, config))
```

Comment:
Verify what is un-serializable with closure serializer; comment on why were are KryoWrapping